### PR TITLE
Clean up network packet handling

### PR DESCRIPTION
**Pull request by valarnin**
This PR has now been tested (see https://github.com/OverlayPlugin/OverlayPlugin/pull/345#issuecomment-2134314178) and is ready for a full review and merge.

The goals of this PR are as follows:

1. Remove boilerplate where possible
2. Ensure that all network packet handling accounts for potential region differences
3. Implement more robust error logging

### CODE REVIEW

**Comment by valarnin** [comment]
@wexxlee I think this is ready for some initial thoughts. Maybe @xpdota as well, since you've written some stuff for network packet handling.

**Comment by xpdota** [comment]
Looks like some good improvements to me.

File: OverlayPlugin.Core/NetworkProcessors/LineActorControlSelfExtra.cs [hunk 1]
```diff
@@ -38,146 +36,87 @@ public class LineActorControlSelfExtra
...
+                var category = Get<Server_ActorControlCategory>("category");
+
+                if (!AllowedActorControlCategories.Contains(category)) return null;
+
+                var param1 = Get<UInt32>("param1");
+                var param2 = Get<UInt32>("param1");
+                var param3 = Get<UInt32>("param1");
+                var param4 = Get<UInt32>("param1");
+                var param5 = Get<UInt32>("param1");
+                var param6 = Get<UInt32>("param1");
```
Comment (wexxlee) [inline OverlayPlugin.Core/NetworkProcessors/LineActorControlSelfExtra.cs:52]: Copy-paste miss on the param names?  i.e., shouldn't it be:
```var param2 = Get<UInt32>("param2");``` etc.

**Review by wexxlee — COMMENTED** [review]
LGTM as well.  Haven't done side-by-side testing to confirm (and I don't have a kr/cn client to test against), but approach-wise, this seems like a nice bit of cleanup.  Just one comment on `ActorControlSelfExtra`.

**Comment by valarnin** [comment]
Alright, this is technically ready for general review. I still need to do a bunch of testing, but the overall refactoring is complete.

**Comment by valarnin** [comment]
I have tested:

- LineAbilityExtra
  - Fixed a few issues with memory and pointers, as well as struct offsets
- LineActorCastExtra, no issues
- LineActorControlExtra & LineActorControlSelfExtra
  - For some reason `string.Format` was throwing an exception due to `:X8` and `:X4`. No clue why, it didn't do that before the refactor, I blame Microsoft. Changed it to just use `$"{...:X8}"` instead.
- LineActorMove, no issues
- LineBattleTalk2
- LineNpcYell
- LineCountdown
- LineCountdownCancel

Still needs tested:
- LineActorSetPos
- LineCEDirector
- LineFateControl
- LineMapEffect
- LineRSV (we can't really test this currently, since there aren't any RSVs left in the game until probably the first savages of Dawntrail)
- LineSpawnNPCExtra

**Comment by wexxlee** [comment]
Next couple of days are pretty crazy for me, but I should have some time Fri/Sat to give this a run.

**Review by wexxlee — APPROVED** [review]
Gave this a workout, and looks good.  This is a nice bit of cleanup & bug fixes.

**Pull request merged**